### PR TITLE
Topic recognition- ADNI Transformer (48245764)

### PR DESCRIPTION
**Pull request by JivalX**
## Problem Statement
Classify Alzheimer’s disease (normal and AD) of the ADNI brain data using
a visual transformer set having a minimum accuracy of 0.8 on the test set.

## Contents

1. modules.py- containing components of the model for classifying Alzheimer's disease
2. dataset.py- containing the data loader for loading and preprocessing your data
3. train.py- file for training, validating, testing, and saving your model. Import the model components from modules.py and the data loader from dataset.py
4. predict.py- demonstrates how to use the trained model. Includes code that makes predictions and provides example outputs.
5. README.md- This file will serve as the project's documentation, and it's essential for providing an overview of the project, dependancies and objectives.

- Used tensorflow for the project

Thank you!
Jival Kapoor (@JivalX)
UQ ID- s4824576

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:
Excellent

Commit messages: Excellent, detailed.

Code:
- Uses Vision transformer
- Well commented
- Good model design

Functionality/Performance:
- Test ACC: 67.2% 
- Preprocssing data correctly
- Model is underfitting, could train more.
- No evidence shows patient level split.
- Well done

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, need patient level split -1
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good, no usage -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required move readme and restore repo README -2
Request Description good

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.
